### PR TITLE
Fix index out of range error when regions config arg is not set 

### PR DESCRIPTION
**Pull request by bigdatasourav**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Add passing integration test logs here
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
Add example SQL query results here (please include the input queries as well)
```
</details>

### CODE REVIEW

File: oci/multi_region.go [hunk 1]
```diff
@@ -182,12 +180,17 @@ func listAllCompartments(ctx context.Context, d *plugin.QueryData) ([]identity.C
...
+	if GetConfig(d.Connection).Regions != nil {
+		endpointRegion = GetConfig(d.Connection).Regions[0]
+	}
+
 	pagesLeft := true
 	for pagesLeft {
 		response, err := session.IdentityClient.ListCompartments(ctx, request)
 		if err != nil {
-			if strings.Contains(err.Error(), "no such host") {
+			if strings.Contains(err.Error(), "no such host") && endpointRegion != "" {
```
Comment (misraved) [inline oci/multi_region.go:193]: @bigdatasourav why are we adding the `endpointRegion != ""` check? Is it necessary?

**Comment by misraved** [comment]
@bigdatasourav could you please add in example query results as well?

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@bigdatasourav Please see comments, thanks!

File: oci/multi_region.go [hunk 2]
```diff
@@ -182,12 +180,17 @@ func listAllCompartments(ctx context.Context, d *plugin.QueryData) ([]identity.C
 		},
 	}
 
-	endpointRegion := GetConfig(d.Connection).Regions[0]
+	// fetch the first region provided in config if available
+	var endpointRegion string
+	if GetConfig(d.Connection).Regions != nil {
```
[the remark this answers is not in the document]
Comment (cbruno10) [reply oci/multi_region.go:185]: For reference, the AWS plugin uses the "default" region to describe the regions (https://github.com/turbot/steampipe-plugin-aws/blob/fd736d30b83ccf7b9db4e6be45fd1deb86db08de/aws/multi_region.go#L446), I think we should do something similar here

**Review by misraved — APPROVED** [review]

**Pull request merged**